misc/scripts/file_format.sh

```diff
@@ -65,18 +65,19 @@ if [ -s utf8-validation.txt ]
 then
     # If the file has content and is not empty, violations
     # detected, notify the user, clean up, and exit.
-    printf "\n*** The following files contain invalid UTF-8 character sequences:\n\n"
+    printf "\n\e[1;33m*** The following files contain invalid UTF-8 character sequences:\e[0m\n\n"
     cat utf8-validation.txt
 fi
 
 rm -f utf8-validation.txt
 
 if [ ! -z "$diff" ]
 then
-    printf "\n*** The following differences were found between the code "
-    printf "and the formatting rules:\n\n"
-    echo "$diff"
+    # A diff has been created, notify the user, clean up, and exit.
+    printf "\n\e[1;33m*** The following changes must be made to comply with the formatting rules:\e[0m\n\n"
+    # Perl commands replace trailing spaces with `·` and tabs with `<TAB>`.
+    printf "$diff\n" | perl -pe 's/(.*[^ ])( +)(\e\[m)$/my $spaces="·" x length($2); sprintf("$1$spaces$3")/ge' | perl -pe 's/(.*[^\t])(\t+)(\e\[m)$/my $tabs="<TAB>" x length($2); sprintf("$1$tabs$3")/ge'
```

Comment:
This was fun (not). If someone has a deep love for regex and sed/perl/awk/bash, feel free to try to optimize this so it's faster, simpler, more reliable, and more readable.